packages/app-desktop/commands/convertNoteToMarkdown.ts

```diff
@@ -0,0 +1,45 @@
+import { _ } from '@joplin/lib/locale';
+import Note from '@joplin/lib/models/Note';
+import { stateUtils } from '@joplin/lib/reducer';
+import { CommandRuntime, CommandDeclaration } from '@joplin/lib/services/CommandService';
+import { MarkupToHtml } from '@joplin/renderer';
+import { runtime as convertHtmlToMarkdown } from '@joplin/lib/commands/convertHtmlToMarkdown';
+
+export const declaration: CommandDeclaration = {
+	name: 'convertNoteToMarkdown',
+	label: () => _('Convert it to Markdown'),
+};
+
+export const runtime = (): CommandRuntime => {
+	return {
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
+		execute: async (context: any, noteId: string = null) => {
+			noteId = noteId || stateUtils.selectedNoteId(context.state);
+
+			const note = await Note.load(noteId);
+
+			const markdownBody = await convertHtmlToMarkdown().execute(context, note.body);
+
+			const newNote = await Note.save({
+				...note,
+				id: undefined,
+				body: markdownBody,
+				markup_language: MarkupToHtml.MARKUP_LANGUAGE_MARKDOWN,
+				user_updated_time: new Date().getTime(),
+			});
+			await Note.delete(note.id, { toTrash: true });
```

Comment:
What happens if there's an error somewhere? Does the UI display some kind of message box?

Reply:
Right now if it fails anywhere, there is no feedback, nothing happens, only the error message is logged. Maybe we could have 3 states for `NOTE_HTML_TO_MARKDOWN_DONE`: `idle`, `success` and `error` in case something goes wrong, what do you think?